op-challenger/fault/monitor_test.go

```diff
@@ -103,10 +105,76 @@ func TestMonitorOnlyCreateSpecifiedGame(t *testing.T) {
 	require.Equal(t, 1, games.created[addr2].progressCount)
 }
 
+func TestMonitorGames(t *testing.T) {
+	t.Run("subscription errors", func(t *testing.T) {
+		addr1 := common.Address{0xaa}
+		addr2 := common.Address{0xbb}
+		monitor, source, games, _, mockSub := setupMonitorTest(t, []common.Address{})
+
+		source.games = []FaultDisputeGame{
+			{
+				Proxy:     addr1,
+				Timestamp: 9999,
+			},
+			{
+				Proxy:     addr2,
+				Timestamp: 9999,
+			},
+		}
+
+		var err error
+		go func() {
+			err = monitor.MonitorGames(context.Background())
+		}()
+		mockSub.errChan <- ethereum.NotFound
+		require.ErrorIs(t, err, ethereum.NotFound)
```

Comment:
Also, this is a race condition depending on how quickly the go func winds up assigning to err.  I think you need a WaitGroup to allow the go func to flag when all its work is done.

Reply:
Good callout hadn't paid attention to that.